custom/ewsghana/reports/specific_reports/reporting_rates.py

```diff
@@ -274,33 +279,51 @@ class AlertsData(ReportingRatesData):
     def headers(self):
         return []
 
+    @property
+    @memoized
+    def locations_ids(self):
+        locations_ids = list(self.location.get_descendants().filter(location_type__administrative=False)
+                             .values_list('location_id', flat=True))
+        return locations_ids
+
     def supply_points_users(self):
+        locations_ids = self.locations_ids
         query = UserES().mobile_users().domain(self.config['domain']).term(
-            "location_id", list(self.config['reporting_supply_points'])
+            "location_id",
+            locations_ids
         )
         with_reporters = set()
-        with_in_charge = set()
 
         for hit in query.run().hits:
             with_reporters.add(hit['location_id'])
-            if 'In Charge' in hit['user_data'].get('role', []):
-                with_in_charge.add(hit['location_id'])
+
+        with_in_charge = set(FacilityInCharge.objects.filter(location__location_id__in=locations_ids)
+                             .values_list('location__location_id', flat=True).distinct())
 
         return with_reporters, with_in_charge
 
     @property
+    @memoized
+    def last_month_reporting_sp_ids(self):
+        return StockState.objects.filter(
+            last_modified_date__gte=datetime.utcnow() - timedelta(days=30),
+            sql_location__location_id__in=self.locations_ids
+        ).values_list('case_id', flat=True)
+
+    @property
     def rows(self):
         rows = []
         if self.location_id:
             supply_points = self.get_supply_points()
             with_reporters, with_in_charge = self.supply_points_users()
+            last_month_reporting_sp_ids = self.last_month_reporting_sp_ids
             for sp in supply_points:
                 url = make_url(
                     StockLevelsReport, self.config['domain'], '?location_id=%s&startdate=%s&enddate=%s',
                     (sp.location_id, json_format_date(self.config['startdate']),
                      json_format_date(self.config['enddate']))
                 )
-                if sp.location_id not in self.config['reporting_supply_points']:
+                if sp.supply_point_id not in last_month_reporting_sp_ids:
                     rows.append(['<div style="background-color: rgba(255, 0, 0, 0.2)">%s has not reported last '
```

Comment:
ok thanks - just a couple things:

- i think we should update the alert message to say "%s has not reported in the last 32 days" rather than "%s has not reported last month" since i think that might have caused some confusion

- it's hard for me to tell - does this report show all the alerts Mike mentioned and does it show them appropriately based on the location level that's being reported on?